data-prepper-api/src/main/java/org/opensearch/dataprepper/model/buffer/AbstractBuffer.java

```diff
@@ -71,8 +71,10 @@ public void write(T record, int timeoutInMillis) throws TimeoutException {
 
         try {
             doWrite(record, timeoutInMillis);
-            recordsWrittenCounter.increment();
-            recordsInBuffer.incrementAndGet();
+            if (!isByteBuffer()) {
+                recordsWrittenCounter.increment();
```

Comment:
With protobuf wrapper over byteBuffer can we store numRecords along with byteBuffer? and then increment/decrement these metrics? these are good metrics for customers IMO.

Reply:
Maybe. I am not sure if it's always possible because it means we have to parse the input, which we do not plan to do afaik.

Reply:
We should add some metrics for the byte counts.

Also, I think we should move away from this `AbstractBuffer`. We make our second buffer and it is already producing erroneous results. :) 

But, that can come later.